src/base/emu-i386/simx86/codegen-sim.c

```diff
@@ -3197,10 +3198,9 @@ static unsigned int _Sim_helper(unsigned int mem_ref, unsigned int data, int mod
 				    if (debug_level('e')>1)
 					e_printf("Return for STI fl=%08x\n",
 						 EFLAGS);
-				    if (opc == POPF)
-					TheCPU.err = (is_tf ? EXCP01_SSTP : EXCP_STISIGNAL);
-				    else
-					data = (is_tf ? EXCP01_SSTP : EXCP_STISIGNAL);
+				    TheCPU.err = (is_tf ? EXCP01_SSTP : EXCP_STISIGNAL);
+				    if (opc == IRET)
+					TheCPU.err = -TheCPU.err; /* avoid early exit */
```

Comment:
But don't we already have
plenty of return indicators,
like CeSxx, flags and so on?
Do you really need to do this?
Its a bit messy.

Reply:
Maybe I can set CeS_xxx in TheCPU.exit_pending instead. Let me try that. These psuedo-exceptions are a bit odd indeed.